mock_services/rules.py

```diff
@@ -0,0 +1,94 @@
+# -*- coding: utf-8 -*-
+import logging
+import re
+
+from functools import partial
+
+import responses
+
+from . import service
+from . import storage
+
+
+logger = logging.getLogger(__name__)
+
+
+def reset_rules():
+    responses.reset()
+    storage.reset()
+
+
+def update_http_rules(rules, content_type='application/json'):
+    """Adds rules to global responses mock.
+
+    It permits to set mock in a more global way than decorators, cf.:
+    https://github.com/getsentry/responses
+
+    Here we assume urls in the passed dict are regex we recompile before adding
+    a rule.
+
+    Rules example:
+
+    >>> def fake_duckduckgo_cb(request):
+    ...     return 200, {}, 'Coincoin!'
+
+    >>> rules = [
+        {
+            'body': 'I am watching you',
+            'method': 'GET',
+            'status': 200,
+            'url': r'^https://www.google.com/#q='
+        },
+        {
+            'callback': fake_duckduckgo_cb,
+            'method': 'GET',
+            'url': r'^https://duckduckgo.com/?q='
+        },
+    ]
+
+    """
+    for i, kw in enumerate(rules):
+
+        if kw['method'] not in ['GET', 'POST', 'PATCH', 'DELETE']:
```

Comment:
It's just a good practice, checking presence in a set is a lot faster. Not that it will have a big effect here, but it's a good habit to catch.